src/libbson/tests/test-bson-vector.c

```diff
@@ -0,0 +1,1581 @@
+         bson_iter_t iter;
+         BSON_ASSERT (bson_iter_init_find (&iter, &vector_doc, "vector"));
+         BSON_ASSERT (bson_vector_int8_const_view_from_iter (&view, &iter));
+         BSON_ASSERT (bson_vector_int8_const_view_read (view, actual_elements, element_count, offset));
+         for (size_t i = 0; i < element_count; i++) {
+            BSON_ASSERT (actual_elements[i] == expected_elements[i + offset]);
+         }
+      }
+   }
+   bson_destroy (&vector_doc);
+   bson_free (expected_elements);
+   bson_free (actual_elements);
+}
+
+static void
+test_bson_vector_view_api_fuzz_float32 (void)
+{
+   size_t current_length = 0;
+   bson_t vector_doc = BSON_INITIALIZER;
+   float *expected_elements = bson_malloc (MAX_TESTED_VECTOR_LENGTH * sizeof *expected_elements);
+   float *actual_elements = bson_malloc (MAX_TESTED_VECTOR_LENGTH * sizeof *actual_elements);
+   for (int fuzz_iter = 0; fuzz_iter < FUZZ_TEST_ITERS; fuzz_iter++) {
+      unsigned r = (unsigned) rand ();
+      unsigned r_operation = r & 0xFu;
+      size_t r_param = r >> 4;
+
+      if (current_length == 0 || r_operation == 15) {
+         // Resize and fill
+         size_t new_length = (size_t) r_param % MAX_TESTED_VECTOR_LENGTH;
+         bson_reinit (&vector_doc);
+         bson_vector_float32_view_t view;
+         BSON_ASSERT (BSON_APPEND_VECTOR_FLOAT32_UNINIT (&vector_doc, "vector", new_length, &view));
+         for (size_t i = 0; i < new_length; i++) {
+            expected_elements[i] = (float) rand ();
+         }
+         BSON_ASSERT (bson_vector_float32_view_write (view, expected_elements, new_length, 0));
+         current_length = new_length;
+
+      } else if (r_operation < 7) {
+         // Partial write
+         size_t element_count = r_param % current_length;
+         size_t offset = rand () % (current_length - element_count);
+         for (size_t i = 0; i < element_count; i++) {
+            expected_elements[offset + i] = (float) rand ();
+         }
+         bson_vector_float32_view_t view;
+         bson_iter_t iter;
+         BSON_ASSERT (bson_iter_init_find (&iter, &vector_doc, "vector"));
+         BSON_ASSERT (bson_vector_float32_view_from_iter (&view, &iter));
+         BSON_ASSERT (bson_vector_float32_view_write (view, expected_elements + offset, element_count, offset));
+
+      } else {
+         // Partial read
+         size_t element_count = r_param % current_length;
+         size_t offset = rand () % (current_length - element_count);
+         bson_vector_float32_const_view_t view;
+         bson_iter_t iter;
+         BSON_ASSERT (bson_iter_init_find (&iter, &vector_doc, "vector"));
+         BSON_ASSERT (bson_vector_float32_const_view_from_iter (&view, &iter));
+         BSON_ASSERT (bson_vector_float32_const_view_read (view, actual_elements, element_count, offset));
+         for (size_t i = 0; i < element_count; i++) {
+            BSON_ASSERT (actual_elements[i] == expected_elements[i + offset]);
```

Comment:
```suggestion
            ASSERT_EQUAL_DOUBLE (actual_elements[i], expected_elements[i + offset]);
```
Dedicated test assertion macro + addresses `-Wfloat-equal` warnings.

On the other hand, `ASSERT_EQUAL_DOUBLE`'s allowance of up to 20% error (according to its failure message) may be a bit much... 🤔